kv/db.go

```diff
@@ -163,6 +165,71 @@ func (s *rpcDBServer) executeCmd(args proto.Request, reply proto.Response) error
 	return nil
 }
 
+func (s *rpcDBServer) CreateTable(args *proto.CreateTableRequest, reply *proto.CreateTableResponse) error {
+	schema := args.Schema
+	tableDesc := proto.TableDescriptor{
+		Id:    0, // allocate here
+		Table: schema.Table,
+	}
+
+	var columnsByName map[string]uint32
+
+	for _, column := range schema.Columns {
+		columnDesc := proto.ColumnDescriptor{
+			Id:     tableDesc.NextColumnId,
+			Column: column,
+		}
+
+		columnsByName[column.Name] = columnDesc.Id
+
+		tableDesc.Columns = append(tableDesc.Columns, columnDesc)
+		tableDesc.NextColumnId++
+	}
+
+	for _, index := range schema.Indexes {
+		indexDesc := proto.IndexDescriptor{
+			Id:    tableDesc.NextIndexId,
+			Index: index.Index,
+		}
+		for _, columnName := range index.ColumnNames {
+			columnID, ok := columnsByName[columnName]
+			if !ok {
+				// return error, bad request
+			}
+			indexDesc.ColumnIds = append(indexDesc.ColumnIds, columnID)
+		}
+
+		tableDesc.Indexes = append(tableDesc.Indexes)
+		tableDesc.NextIndexId++
+	}
+
+	bytes, err := tableDesc.Marshal()
+	if err != nil {
+		return err
+	}
+
+	resp := &proto.ConditionalPutResponse{}
```

Comment:
Server internal code generally uses the `client.DB` interface when acting as a client. That you're not doing here makes the code more awkward than necessary. Creating a `client.DB` from a `client.Sender` is not difficult:

```
  db, err := client.Open("//root@", client.SenderOpt(sender))
```